core/client_test.go

```diff
@@ -102,3 +103,21 @@ func TestRemoteClient_StartBlockSubscription_And_GetBlock(t *testing.T) {
 	require.NoError(t, client.Stop())
 	require.NoError(t, remote.Stop())
 }
+
+// TestRemoteClient_RetryDial ensures 3 additional attempts
+// are made to dial the remote node.
+func TestRemoteClient_RetryDial(t *testing.T) {
+	remote := StartMockNode()
+	protocol, ip := getRemoteEndpoint(remote)
+
+	// deliberately stop remote
+	err := remote.Stop()
+	require.NoError(t, err)
+
+	client, err := NewRemote(protocol, ip)
+	require.NoError(t, err)
+
+	_, err = client.Block(context.Background(), nil)
+	require.Error(t, err)
+	assert.ErrorContains(t, err, "giving up after 3 attempt(s)")
```

Comment:
Can't this be tested against the error type itself instead of via string? (if it is part of their public API)

Reply:
They don't have err types